### PR TITLE
Matching Algorithm

### PR DESCRIPTION
**Pull request by Joeyh021**
3 tests cover the basics of the features of it and the system, more would certainly be welcome.

### CODE REVIEW

**Review by NoahNok — APPROVED** [review]
Oooooh ai vibes :)

File: people/matching.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+from people.models import *
+from events.models import Event
+
+
+def get_matches(mentee: User):
+    """
+    given the uuid of a mentee, find the best possible mentors, returning a list of their uuids in order of suitability
+    if debug, then a dict of the calculated values is returned for testing/debugging purposes
+    """
+    mentor_type = UserType.objects.get(type="mentor")
```
Comment (joshdavies14) [inline people/matching.py:10]: Once my changes with switching this to an enum are merged we'll need to change this slightly, not major for now though!

File: tests/test_data.py [hunk 2]
```diff
@@ -1,48 +1,213 @@
...
 def create_data() -> None:
 
     # create user types
-    mentee_type = models.UserType.objects.create(id=uuid4(), type="mentee")
-    mentor_type = models.UserType.objects.create(id=uuid4(), type="mentor")
-    mentor_mentee_type = models.UserType.objects.create(
-        id=uuid4(), type="mentor mentee"
-    )
-    none_usertype = models.UserType.objects.create(id=uuid4(), type="")
+    mentee_type = UserType.objects.create(type="mentee")
```
Comment (joshdavies14) [inline tests/test_data.py:8]: Same here with changing once my changes are merged, don't worry for now though :P

**Review by joshdavies14 — APPROVED** [review]
All LGTM!

Also don't haunt me with AI :P

**Pull request merged**